From f21acdccfdc820e60f91209a717475cfbbbf7382 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 19 Mar 2019 15:19:10 +0200
Subject: [PATCH 3/5] MLK-21190 arm64: dts: imx8mq: fix DTS warnings

commit 15ccbb77f0a687e738ba65979108dd36cb45d0f7 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch fixes the folowing DTS warnings:

 Warning (graph_endpoint): /dcss@0x32e00000/port@0/hdmi-endpoint: graph endpont node name should be 'endpoint'
 Warning (graph_endpoint): /dcss@0x32e00000/port@0/mipi_dsi: graph endpont node name should be 'endpoint'
 Warning (graph_endpoint): /lcdif@30320000/port@0/mipi-dsi-endpoint: graph endpont node name should be 'endpoint'
 Warning (graph_endpoint): /mipi_csi1@30a70000/port/endpoint1: graph endpont node name should be 'endpoint'
 Warning (graph_endpoint): /mipi_csi1@30a70000/port/endpoint2: graph endpont node name should be 'endpoint'
 Warning (graph_endpoint): /mipi_csi2@30b60000/port/endpoint1: graph endpont node name should be 'endpoint'
 Warning (graph_endpoint): /mipi_csi2@30b60000/port/endpoint2: graph endpont node name should be 'endpoint'
 Warning (interrupts_property): /dcss@0x32e00000:#interrupt-cells: size is (64), expected multiple of 12

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
---
 .../boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi   | 8 ++++++--
 .../boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi   | 8 ++++++--
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts   | 2 ++
 .../boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi  | 8 +++++++-
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts          | 8 ++++----
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi             | 8 ++++----
 6 files changed, 29 insertions(+), 13 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
index 77a612bd6a1b..b3ed6893c696 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
@@ -18,8 +18,12 @@
 	};
 };
 
+
+/delete-node/ &hdmi_port;
+
 &hdmi {
 	status = "disabled";
+
 };
 
 &dcss {
@@ -43,10 +47,10 @@
 			       <800000000>,
 			       <400000000>;
 
-	dcss_disp0: port@0 {
+	dcss_port: port@0 {
 		reg = <0>;
 
-		dcss_disp0_mipi_dsi: mipi_dsi {
+		dcss_disp0_mipi_dsi: endpoint {
 			remote-endpoint = <&mipi_dsi_in>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
index d801dba4d92e..8eb08692cf30 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
@@ -18,8 +18,12 @@
 	};
 };
 
+
+/delete-node/ &hdmi_port;
+
 &hdmi {
 	status = "disabled";
+
 };
 
 &dcss {
@@ -49,10 +53,10 @@
 			       <400000000>,
 			       <599999999>;
 
-	dcss_disp0: port@0 {
+	dcss_port: port@0 {
 		reg = <0>;
 
-		dcss_disp0_mipi_dsi: mipi_dsi {
+		dcss_disp0_mipi_dsi: endpoint {
 			remote-endpoint = <&mipi_dsi_in>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts
index 0914012d016d..62f0e27ce8f2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-inmate.dts
@@ -160,6 +160,8 @@
 
 &dcss {
 	/delete-property/ interrupt-parent;
+	/delete-property/ interrupts;
+	/delete-property/ interrupt-names;
 };
 
 /delete-node/ &gpc;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi
index c93885da4d55..1aa04548d23b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi
@@ -15,6 +15,8 @@
 / {
 	display-subsystem {
 		status = "disabled";
+
+		/delete-property/ ports;
 	};
 
 	sound-hdmi {
@@ -26,10 +28,14 @@
 	status = "disabled";
 };
 
+/delete-node/ &dcss_port;
+
 &dcss {
 	status = "disabled";
 };
 
+/delete-node/ &hdmi_port;
+
 &hdmi {
 	status = "disabled";
 };
@@ -39,7 +45,7 @@
 	max-res = <1280>, <720>;
 
 	port@0 {
-		lcdif_mipi_dsi: mipi-dsi-endpoint {
+		lcdif_mipi_dsi: endpoint {
 			remote-endpoint = <&mipi_dsi_in>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index a9d93cb67324..f804adb49df7 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -958,12 +958,12 @@
 	#size-cells = <0>;
 	status = "okay";
 	port {
-		mipi1_sensor_ep: endpoint1 {
+		mipi1_sensor_ep: endpoint@0 {
 			remote-endpoint = <&ov5640_mipi1_ep>;
 			data-lanes = <1 2>;
 		};
 
-		csi1_mipi_ep: endpoint2 {
+		csi1_mipi_ep: endpoint@1 {
 			remote-endpoint = <&csi1_ep>;
 		};
 	};
@@ -974,12 +974,12 @@
 	#size-cells = <0>;
 	status = "okay";
 	port {
-		mipi2_sensor_ep: endpoint1 {
+		mipi2_sensor_ep: endpoint@0 {
 			remote-endpoint = <&ov5640_mipi2_ep>;
 			data-lanes = <1 2>;
 		};
 
-		csi2_mipi_ep: endpoint2 {
+		csi2_mipi_ep: endpoint@1 {
 			remote-endpoint = <&csi2_ep>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index f4dcf7ac3c98..a461365cec79 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -566,10 +566,10 @@
 				       <400000000>;
 		status = "disabled";
 
-		dcss_disp0: port@0 {
+		dcss_port: port@0 {
 			reg = <0>;
 
-			dcss_disp0_hdmi: hdmi-endpoint {
+			dcss_disp0_hdmi: endpoint {
 				remote-endpoint = <&hdmi_disp>;
 			};
 		};
@@ -588,7 +588,7 @@
 		fsl,cec;
 		status = "disabled";
 
-		port@0 {
+		hdmi_port: port@0 {
 			reg = <0>;
 			hdmi_disp: endpoint {
 				remote-endpoint = <&dcss_disp0_hdmi>;
@@ -610,7 +610,7 @@
 
 	display-subsystem {
 		compatible = "fsl,imx-display-subsystem";
-		ports = <&dcss_disp0>;
+		ports = <&dcss_port>;
 	};
 
 	mipi_dsi_phy: dsi_phy@30A00300 {
-- 
2.17.1

